File: pyton_files/ex9.py
from abc import ABC, abstractmethod


class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass


class Relatorio(Imprimivel):
    def __init__(self, conteudo):
        self.conteudo = conteudo

    def imprimir(self):
        print(f"Imprimindo Relatório: {self.conteudo}")


class Contrato(Imprimivel):
    def __init__(self, partes):
        self.partes = partes

    def imprimir(self):
        print(f"Imprimindo Contrato entre: {', '.join(self.partes)}")


def main():
    relatorio = Relatorio("Análise de Desempenho 2024")
    contrato = Contrato(["Empresa A", "Empresa B"])

    
    relatorio.imprimir()
    contrato.imprimir()


if __name__ == "__main__":
    main()

File: go_files/ex9.go
package main

import "fmt"

type Imprimivel interface {
	Imprimir()
}

type Relatorio struct {
	conteudo string
}

func (r Relatorio) Imprimir() {
	fmt.Println("Imprimindo Relatório:", r.conteudo)
}

type Contrato struct {
	detalhes string
}

func (c Contrato) Imprimir() {
	fmt.Println("Imprimindo Contrato:", c.detalhes)
}

func main() {
	var itens []Imprimivel
	itens = append(itens, Relatorio{conteudo: "Relatório de Vendas"})
	itens = append(itens, Contrato{detalhes: "Contrato de Trabalho"})

	for _, item := range itens {
		item.Imprimir()
	}
}
